cfgov/teachers_digital_platform/models/activity_index_page.py

```diff
@@ -0,0 +1,592 @@
+        )
+        payload = {
+            'search_query': search_query,
+            'results': results,
+            'total_results': total_results,
+            'total_activities': total_activities,
+            'selected_facets': selected_facets,
+            'all_facets': all_facets,
+        }
+        # List all facet blocks that need to be expanded
+        conditionally_expanded = {
+            facet_name for facet_name, facet_items in all_facets.items()
+            if any(
+                facet['selected'] is True for facet in facet_items
+            )
+        }
+        expanded_facets = ALWAYS_EXPANDED.union(set(conditionally_expanded))
+        payload.update({
+            'expanded_facets': expanded_facets,
+        })
+        self.results = payload
+        results_per_page = validate_results_per_page(request)
+        paginator = Paginator(payload['results'], results_per_page)
+        current_page = validate_page_number(request, paginator)
+        paginated_page = paginator.page(current_page)
+        context_update = {
+            'facets': all_facets,
+            'activities': paginated_page,
+            'total_results': total_results,
+            'results_per_page': results_per_page,
+            'current_page': current_page,
+            'paginator': paginator,
+            'show_filters': bool(selected_facets),
+        }
+        return context_update
+
+    # TODO: Remove this function when we switch to DSL
+    def haystack_search(self, request, *args, **kwargs):
+        search_query = request.GET.get('q', '')  # haystack cleans this string
+        sqs = SearchQuerySet().models(ActivityPage)
+        total_activities = sqs.count()
+        # Load selected facets
+        selected_facets = {}
+        facet_queries = {}
+        activity_card_map = self.activity_setups.card_setup
+
+        for facet, facet_config in FACET_MAP:
+            sqs = sqs.facet(str(facet), size=facet_config[2])
+            if facet in request.GET and request.GET.get(facet):
+                selected_facets[facet] = [
+                    int(value) for value in request.GET.getlist(facet)
+                    if value.isdigit()
+                ]
+                facet_queries[facet] = facet + '_exact:' + (
+                    " OR " + facet + "_exact:").join(
+                    [str(value) for value in selected_facets[facet]]
+                )
+        payload = {
+            'search_query': search_query,
+            'results': [],
+            'total_results': 0,
+            'total_activities': total_activities,
+            'selected_facets': selected_facets,
+            'facet_queries': facet_queries,
+            'all_facets': {},
+        }
+
+        # Apply search query if it exists, but don't apply facets
+        if search_query:
+            sqs = sqs.filter(
+                content=search_query).order_by('-_score', '-date')
+        else:
+            sqs = sqs.order_by('-date')
```

Comment:
This line isn't covered by a test.

Reply:
Ditto from above. This has never been covered by tests, and now is not the time to add them.